src/api/private/auth/auth.controller.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  NotFoundException,
+  Post,
+  Put,
+  UseGuards,
+} from '@nestjs/common';
+import { RegisterDto } from '../../../identity/internal/register.dto';
+import { ConsoleLoggerService } from '../../../logger/console-logger.service';
+import { UsersService } from '../../../users/users.service';
+import { IdentityService } from '../../../identity/identity.service';
+import { ChangePasswordDto } from '../../../identity/internal/change-password.dto';
+import { FunctionalityDisabled, NotInDBError } from '../../../errors/errors';
+import { LoginDto } from '../../../identity/internal/login.dto';
+import { LocalAuthGuard } from '../../../identity/internal/local-auth.guard';
+
+@Controller('auth')
+export class AuthController {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    private usersService: UsersService,
+    private identityService: IdentityService,
+  ) {
+    this.logger.setContext(AuthController.name);
+  }
+
+  @Post('internal')
+  async registerUser(@Body() registerDto: RegisterDto): Promise<void> {
+    try {
+      const user = await this.usersService.createUser(
+        registerDto.username,
+        registerDto.username,
```

Comment:
I guess we wanted to separate username from display name in 2.x.
At least we built that in the frontend a year ago.

Reply:
In principle it's seperated, but the current frontend implementation only asks for a username and not for a seperate display name so I kept with that. In principle on could add a display name to the RegisterDto and change those lines. 
![grafik](https://user-images.githubusercontent.com/3978662/128753024-2c46fd76-e4d3-444d-a790-21343fe58796.png)

Reply:
I added the displayname to the RegisterDto just now